### DISCUSSION TITLE
When Linux 86-64x version?

### DISCUSSION THREAD

**Discussion by LeFFaQ** [cherryleafroad/kmagick-rs#17] [Q&A] [answered]
I want to build a simple web server to converting images for home use on linux machine. I follow "android" instruction, but it not work with ``` java.lang.UnsatisfiedLinkError: no kmagick in java.library.path: ... ```

**Comment by MolotovCherry** [comment, marked as answer]
Well, in general, assuming you're using Linux x64, you need:
1, kmagick compiled for Linux x64
2, kmagick in your PATH
3, imagemagick installed and in your PATH
(In general, if you're using a desktop OS, your instructions are closer to windows than Android, since android instructions are very specific)

I haven't ever tested this for Linux because I don't have time/use linux atm, but you want, you can try installing `imagemagick-dev` (or equivalent for your system), imagemagick, rust, and compiling it yourself for target `x86_64-unknown-linux-musl`.

I don't believe there's anything in the actual code that's windows specific (obviously it compiles and works for android linux already), so just need the right configuration.

Some methods may be missing or error out since I have compilation checks for specifically android, and default to the windows definitions (so if the generated headers are different, it may cause a compilation issue), not that that'd be super difficult to fix though.

Building instructions for the rust library can be found on the docs page [here](https://github.com/MolotovCherry/kmagick/tree/main/rust). The windows instructions are not all that different from what you'll need to do on linux, just adjust accordingly. The build script will need to be adjusted of course, but the basic idea is there
